marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/savedQueries/saveSavedQuery.sjs

```diff
@@ -0,0 +1,36 @@
+'use strict';
+declareUpdate();
+
+var saveQuery;
+var userCollections = ["http://marklogic.com/data-hub/saved-query"];
+var queryDocument = JSON.parse(saveQuery);
+
+try {
+  if(queryDocument.savedQuery.id) {
+    queryDocument.savedQuery.systemMetadata.lastUpdatedBy = xdmp.getCurrentUser();
+    queryDocument.savedQuery.systemMetadata.lastUpdatedDateTime = fn.currentDateTime();
+  } else {
+    queryDocument.savedQuery.id = sem.uuidString();
+    queryDocument.savedQuery.owner = xdmp.getCurrentUser();
+    queryDocument.savedQuery.systemMetadata = {
+      "createdBy": xdmp.getCurrentUser(),
+      "createdDateTime": fn.currentDateTime(),
+      "lastUpdatedBy": xdmp.getCurrentUser(),
+      "lastUpdatedDateTime": fn.currentDateTime()
+    }
+  }
+
+  let docUri = "/saved-queries/"+queryDocument.savedQuery.id;
+  let permissions = [xdmp.permission('data-hub-saved-query-reader', 'read'),
```

Comment:
Also include "xdmp.defaultPermissions()"

Reply:
Added it in the new commit